src/lib/dialog/dialog-ref.ts

```diff
@@ -14,10 +14,18 @@ export class MdDialogRef<T> {
   /** The instance of component opened into the dialog. */
   componentInstance: T;
 
+  /** Expose overlay backdrop click event */
+  backdropClicked: Observable<void>;
+
+  /** Subject for notifying the user that esc key way pressed */
+  escapePressed = new Subject<void>();
```

Comment:
Alright, it makes sense. I still think that they should be combined into something along the lines of `userGeneratedClose`.